Completed TDD Rspec Palindrome homework #2

daktest opened the pull request:
- Added tests for the Television class

[#152207328]

daktest committed:
Completed TDD Rspec Palindrome homework #2

 - Added tests for the Television class

[#152207328]

jasonnoble commented on objects_tdd/spec/lib/television_spec.rb:
```diff
@@ -0,0 +1,108 @@
+
+  describe '#set_channel(number)' do
+    it 'sets the channel to the number if within valid range' do
+      subject.power
```
If you need to turn the power on before changing the channel, that's a good use for a `before` block:

    before(:each) do
      subject.power
    end

daktest replied:
@jasonnoble There is one test that should not have the power turned on.  How do you exclude a single test from running the before block?

jasonnoble commented on objects_tdd/spec/lib/television_spec.rb:
```diff
@@ -0,0 +1,108 @@
+      subject.power
+      subject.channel_up
+      expect(subject.channel).to eq(4)
+    end
```
What if you hit `channel_up` and you're already on the highest channel?  Write a test that verifies it "rolls" over to the lowest channel.

jasonnoble commented on objects_tdd/spec/lib/television_spec.rb:
```diff
@@ -0,0 +1,108 @@
+      subject.power
+      subject.channel_set(37)
+      subject.channel_down
+      expect(subject.channel).to eq(36)
```
Same comment for lowest channel should "roll" over to the highest channel.

jasonnoble commented on objects_tdd/spec/lib/television_spec.rb:
```diff
@@ -0,0 +1,108 @@
+    end
+  end
+
+  describe 'channel loops when end of range is reached' do
```
This should be inside the `#channel_up` block

jasonnoble commented on objects_tdd/spec/lib/television_spec.rb:
```diff
@@ -0,0 +1,108 @@
+      expect(subject.channel).to eq(3)
+    end
+
+    it '#channel_down loops to last channel after reaching first channel' do
```
Inside the `#channel_down` block.

daktest committed:
Made suggested changes

 - added a `context` block around test that need to call `#power` first
 - added a `before` block to that context
 - moved the channel looping tests into their respective channel_up/down test blocks

[#152207328]
